packages/dev/sharedUiComponents/src/fluent/primitives/chooseTexture.tsx

```diff
@@ -0,0 +1,69 @@
+import type { BaseTexture } from "core/Materials/Textures/baseTexture";
+import type { Scene } from "core/scene";
+import type { Nullable } from "core/types";
+import type { FunctionComponent } from "react";
+import type { PrimitiveProps } from "./primitive";
+
+import { makeStyles, tokens } from "@fluentui/react-components";
+import { useMemo } from "react";
+import { TextureUpload } from "../hoc/textureUpload";
+import { ComboBox } from "./comboBox";
+
+const useStyles = makeStyles({
+    container: {
+        display: "flex",
+        flexDirection: "row",
+        alignItems: "center",
+        gap: tokens.spacingHorizontalS,
+    },
+});
+
+export type ChooseTextureProps = PrimitiveProps<Nullable<BaseTexture>> & {
+    /**
+     * The scene to get textures from
+     */
+    scene: Scene;
+    /**
+     * File types to accept for upload
+     */
+    accept?: string;
+    /**
+     * Whether to only allow cube textures
+     */
+    cubeOnly?: boolean;
+};
+
+/**
+ * A primitive component with a ComboBox for selecting from existing scene textures
+ * and a button for uploading new texture files.
+ * @param props ChooseTextureProps
+ * @returns ChooseTexture component
+ */
+export const ChooseTexture: FunctionComponent<ChooseTextureProps> = (props) => {

```

Comment:
For other similar cases, like selecting a `Material` for `Mesh.activeMaterial` or `Node` for `Node.parent`, how do you imagine this? Do we introduce `ChooseMaterial` and `ChooseNode` components that are similar, but don't have the TextureUpload button?